symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/test/java/com/symphony/bdk/bot/sdk/command/MultiResponseComposerImplTest.java

```diff
@@ -0,0 +1,243 @@
+package com.symphony.bdk.bot.sdk.command;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import com.symphony.bdk.bot.sdk.event.model.MessageAttachmentFile;
+import com.symphony.bdk.bot.sdk.symphony.model.SymphonyMessage;
+
+import model.FileAttachment;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class MultiResponseComposerImplTest {
+
+  private SymphonyMessage symphonyMessage;
+  private MultiResponseComposerImpl multiResponseComposerImpl;
+  private MultiResponseComposerImpl multiResponseComposerImplMock; // to test hasContent()
+
+  @Before
+  public void init() {
+    this.symphonyMessage = this.initSymphonyMessage();
+    this.multiResponseComposerImpl = new MultiResponseComposerImpl();
+    assertTrue(this.multiResponseComposerImpl.isComplete());
+
+    this.multiResponseComposerImplMock = Mockito.mock(MultiResponseComposerImpl.class);
+  }
+
+  @Test
+  public void testCompose() {
+    final ComposerMessageDefinition composerMessageDefinition = this.multiResponseComposerImpl.compose();
+    assertFalse(this.multiResponseComposerImpl.isComplete());
+    assertEquals(composerMessageDefinition, this.multiResponseComposerImpl);
+
+    final Map<SymphonyMessage, Set<String>> expectedComposedResponse = new HashMap<>();
+    assertEquals(expectedComposedResponse, this.multiResponseComposerImpl.getComposedResponse());
+  }
+
+  @Test
+  public void testWithMessage() {
+    final String message = "message1";
+
+    final ComposerAttachmentOrStreamDefinition composerAttachmentOrStreamDefinition =
+        this.multiResponseComposerImpl.withMessage(message);
+    assertEquals(composerAttachmentOrStreamDefinition, this.multiResponseComposerImpl);
```

Comment:
To check that message has been added into `this.multiResponseComposerImpl`, please call `this.multiResponseComposerImpl.toStreams("streamId")` and `this.multiResponseComposerImpl.getComposedResponse()` to check if the map is the expected one.
Please do the same for other methods `with*`.

Reply:
I tried that : 

```
  @Test
  public void testWithMessage() {
    final String message = "message1";

    final ComposerAttachmentOrStreamDefinition composerAttachmentOrStreamDefinition =
        this.multiResponseComposerImpl.withMessage(message);
    assertEquals(composerAttachmentOrStreamDefinition, this.multiResponseComposerImpl);

    this.multiResponseComposerImpl.toStreams(message);
    final Map<SymphonyMessage, Set<String>> composedResponse = this.multiResponseComposerImpl.getComposedResponse();
    assertEquals(message, composedResponse.get(message));
  }
```

but it doesn't work. Could you explain me how to do please ?

Reply:
Like that :
```
  @Test
  public void testWithMessage() {
    final String message = "message1";

    final ComposerAttachmentOrStreamDefinition composerAttachmentOrStreamDefinition =
        this.multiResponseComposerImpl.withMessage(message);
    assertEquals(composerAttachmentOrStreamDefinition, this.multiResponseComposerImpl);

    this.multiResponseComposerImpl.toStreams("streamId");
    final Map<SymphonyMessage, Set<String>> composedResponse = this.multiResponseComposerImpl.getComposedResponse();
    assertEquals(message, composedResponse.get("streamId"));
  }
```
  
  it doesn't work either

Reply:
I realized you have to call the `compose()` method first. This is working:
```java
@Test
  public void testComposeWithMessage() {
    final String message = "message1";
    final String streamId = "streamId";

    this.multiResponseComposerImpl.compose().withMessage(message);
    this.multiResponseComposerImpl.toStreams(streamId);

    final Map<SymphonyMessage, Set<String>> composedResponse = this.multiResponseComposerImpl.getComposedResponse();
    final List<Map.Entry<SymphonyMessage, Set<String>>> entries = new ArrayList(composedResponse.entrySet());

    assertEquals(1, entries.size());

    final Map.Entry<SymphonyMessage, Set<String>> entry = entries.get(0);
    assertEquals(message, entry.getKey().getMessage());
    assertEquals(Collections.singleton(streamId), entry.getValue());
  }
```